python/ray/data/_internal/execution/legacy_compat.py

```diff
@@ -151,7 +151,7 @@ def do_read(blocks: Iterator[Block]) -> Iterator[Block]:
             for read_task in blocks:
                 yield from read_task()
 
-        return MapOperator(do_read, inputs, name="DoRead")
+        return MapOperator.from_compute(do_read, inputs, name="DoRead")
```

Comment:
I think that `.from_compute()` is a bit of a misnomer when `compute` is an optional arg, going to rename this to a more general construct name like `.create()`.